utility/src/main/java/org/oppia/android/util/parser/html/LiTagHandler.kt

```diff
@@ -0,0 +1,173 @@
+
+import android.content.Context
+import android.text.Editable
+import android.text.Spannable
+import android.text.Spanned
+import org.oppia.android.util.locale.OppiaLocale
+import java.util.Stack
+import javax.inject.Inject
+
+/** The custom <li> tag corresponding to [LiTagHandler]. */
+const val CUSTOM_LIST_LI_TAG = "oppia-li"
+
+/** The custom <ul> tag corresponding to [LiTagHandler]. */
+const val CUSTOM_LIST_UL_TAG = "oppia-ul"
+
+/** The custom <ol> tag corresponding to [LiTagHandler]. */
+const val CUSTOM_LIST_OL_TAG = "oppia-ol"
+
+/**
+ * A custom tag handler for properly formatting bullet items in HTML parsed with
+ * [CustomHtmlContentHandler].
+ */
+class LiTagHandler @Inject constructor(
+  private val context: Context,
+  private val displayLocale: OppiaLocale.DisplayLocale
+) :
+  CustomHtmlContentHandler.CustomTagHandler {
+
+  override fun handleOpeningTag(
+    output: Editable,
+    tag: String,
+    lists: Stack<CustomHtmlContentHandler.ListTag>
+  ) {
+    when (tag) {
+      CUSTOM_LIST_UL_TAG ->
+        lists.push(Ul(context, tag))
+
+      CUSTOM_LIST_OL_TAG ->
+        lists.push(Ol(context, tag, displayLocale))
+
+      CUSTOM_LIST_LI_TAG ->
+        lists.peek().openItem(output)
+    }
+  }
+
+  override fun handleClosingTag(
+    output: Editable,
+    indentation: Int,
+    tag: String,
+    lists: Stack<CustomHtmlContentHandler.ListTag>
+  ) {
+    when (tag) {
+      CUSTOM_LIST_UL_TAG ->
+        lists.pop()
+
+      CUSTOM_LIST_OL_TAG ->
+        lists.pop()
+
+      CUSTOM_LIST_LI_TAG ->
+        lists.peek().closeItem(output, indentation = lists.size - 1)
+    }
+  }
+
+  /**
+   * Subclass of [ListTag] for unordered lists.
+   */
+  private class Ul(private val context: Context, private val tag: String) :
+    CustomHtmlContentHandler.ListTag {
+
+    override fun openItem(text: Editable) {
+      appendNewLine(text)
+      start(text, BulletListItem())
+    }
+
+    override fun closeItem(text: Editable, indentation: Int) {
+      appendNewLine(text)
+
+      getLast<BulletListItem>(text)?.let { mark ->
+        setSpanFromMark(text, mark, ListItemLeadingMarginSpan(context, indentation, "•", tag))
+      }
+    }
+  }
+
+  /**
+   * Subclass of [ListTag] for ordered lists.
+   */
+  private class Ol(
+    private val context: Context,
+    private val tag: String,
+    private val displayLocale: OppiaLocale.DisplayLocale
+  ) : CustomHtmlContentHandler.ListTag {
+
+    private var index = 1
+
+    override fun openItem(text: Editable) {
+      appendNewLine(text)
+      start(text, NumberListItem(index))
+      index++
+    }
+
+    override fun closeItem(text: Editable, indentation: Int) {
+      appendNewLine(text)
+
+      getLast<NumberListItem>(text)?.let { mark ->
+        setSpanFromMark(
+          text, mark,
+          ListItemLeadingMarginSpan(
+            context,
+            indentation,
+            "${displayLocale.run {(mark.number).toHumanReadableString(mark.number)}}.",
```

Comment:
```suggestion
            "${displayLocale.run {mark.number.toHumanReadableString(mark.number)}}.",
```

Removing redundant parentheses.